src/lib/storage/dictionary_segment/attribute_vector_iterable.hpp

```diff
@@ -63,6 +63,13 @@ class AttributeVectorIterable : public PointAccessibleSegmentIterable<AttributeV
 
     bool equal(const Iterator& other) const { return _attribute_it == other._attribute_it; }
 
+    void advance(std::ptrdiff_t n) {
+      _attribute_it += n;
+      _chunk_offset += n;
+    }
+
+    std::ptrdiff_t distance_to(const Iterator& other) const { return other._attribute_it - _attribute_it; }
+
```

Comment:
We also started implementing RandomAccessIterators for the Segment Iterators. Good to know we can just copy them from this PR :)
@hendraet @johannes-schneider